crm-app/src/component/customer/Profile.js

```diff
@@ -0,0 +1,179 @@
+import { connect } from 'react-redux';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { addCustomer, updateCustomer, deleteCustomer, resetCustomer } from '../../actions';
+
+function addButton (name, handle) {
+	return (
+		<button type="button" onClick={handle}>
+			{name}
+		</button>
+	);
+}
+
+function enableElement (element) {
+	if (element.disabled) {
+		element.disabled = false;
+	}
+}
+
+function disableElement (element) {
+	if (!element.disabled) {
+		element.disabled = true;
+	}
+}
+
+class CustomerProfile extends Component {
+	constructor (props) {
+		super(props);
+		this.formRef = React.createRef();
+		this.isWritten = false;
+	}
+
+	handleSave = (e) => {
+		const { displayedIndex, onAdd, onUpdate } = this.props;
+		const form = e.target.form;
+		const name = form.name.value;
+		const mail = form.mail.value;
+		const phone = form.phone.value;
+
+		if (name === '' || mail === '' || phone === '') {
+			return;
+		}
+
+		// check if it is new customer or modified customer
+		if (this.isWritten) {
+			onUpdate(displayedIndex, name, phone);
+		} else {
+			if (displayedIndex < 0) {
+				onAdd(name, mail, phone);
+			} else {
+				alert('Error: This email alreay exists.');
+			}
+		}
+
+		e.preventDefault();
+	}
+
+	handleReset = (e) => {
+		this.clearForm(e.target.form);
+		this.props.onReset();
+	}
+
+	handleDelete = (e) => {
```

Comment:
Which one would it be better?
```
1)
	handleDelete  (e) {
		const { displayedIndex, onDelete } = this.props;
		onDelete(displayedIndex);
	}
2)
	handleDelete = (e) => {
		const { displayedIndex, onDelete } = this.props;
		onDelete(displayedIndex);
	}
```

Reply:
2) is better because handleDelete method needs 'this' object to get props data.